nemo/collections/nlp/models/language_modeling/megatron_base_model.py

```diff
@@ -210,7 +211,18 @@ def _reconfigure_val_batches(self):
         Reconfigure trainer.limit_val_batches for pretraining
         """
         # Override limit_val_batches to be a multiple of num microbatches and so there are limit_val_batches//num_micro_batches num of global batches
-        self.trainer.limit_val_batches *= get_num_microbatches()
+        if isinstance(self.trainer.limit_val_batches, int):
+            self.trainer.limit_val_batches *= get_num_microbatches()
+        else:
+            assert isinstance(self.trainer.limit_val_batches, float)
+            if self._validation_ds is not None and self.trainer.limit_val_batches < 1.0:
+                val_length = len(self._validation_ds)
+                if not math.isinf(val_length):
+                    mb_times_dp = get_micro_batch_size() * parallel_state.get_data_parallel_world_size()
+                    total_val_microbatches = val_length // mb_times_dp
+                    limit_val_batches = int(self.trainer.limit_val_batches * total_val_microbatches)
+                    self.trainer.limit_val_batches = limit_val_batches - limit_val_batches % get_num_microbatches()
```

Comment:
@odelalleau small clarification here: why are we subtracting `limit_val_batches` by `limit_val_batches % get_num_microbatches()` ?

Reply:
This is to ensure it is a mutliple of `get_num_microbatches()`, so that we don't process "partial" validation batches.